File: 17/01.hs
import Data.List

parse :: IO [Int]
parse = map (read . pure) . init <$> readFile "01.txt"

rotate :: Int -> [a] -> [a]
rotate = drop <> take

solve :: Int -> [Int] -> Int
solve n xs = sum $ zipWith (\ x y -> if x == y then x else 0) xs (rotate n xs) 

part1, part2 :: [Int] -> Int
part1 = solve 1
part2 xs = solve (length xs `div` 2) xs

main :: IO ()
main = do
  input <- parse
  mapM_ print $ [part1, part2] <*> pure input

File: 21/01.py
a=open('1.txt').read().split('\n')[:-1]
a=list(map(int,a))
b=sum(a[i]>a[i-1]for i in range(1,len(a)))
print(b)
a=[sum(a[i:i+3])for i in range(len(a)-2)]
b=sum(a[i]>a[i-1]for i in range(1,len(a)))
print(b)
